test_conformance/extensions/cl_mutable_dispatch_config_khr/command_buffer_test_mutable_dispatch.cpp

```diff
@@ -0,0 +1,392 @@
+
+    cl_mutable_dispatch_fields_khr mutable_capabilities;
+    CL_CHECK(clGetDeviceInfo(
+        device, CL_DEVICE_MUTABLE_DISPATCH_CAPABILITIES_KHR,
+        sizeof(mutable_capabilities), &mutable_capabilities, nullptr));
+    if (!(mutable_capabilities & CL_MUTABLE_DISPATCH_ARGUMENTS_KHR))
+    {
+        std::cerr << "Device does not support update arguments to a "
+                     "mutable-dispatch, "
+                     "skipping example.\n";
+        return TEST_FAIL;
+    }
+
+    return TEST_PASS;
+}
+
+int query_test_common(cl_device_id& device, cl_context& context,
+                      cl_command_queue& queue,
+                      cl_command_buffer_khr& command_buffer, cl_kernel& kernel)
+{
+    cl_int error = CL_SUCCESS;
+
+    GET_PFN(device, clCreateCommandBufferKHR);
+    GET_PFN(device, clCommandNDRangeKernelKHR);
+    GET_PFN(device, clGetMutableCommandInfoKHR);
+
+    const cl_command_buffer_properties_khr props[] = {
+        CL_COMMAND_BUFFER_FLAGS_KHR,
+        CL_COMMAND_BUFFER_MUTABLE_KHR,
+        0,
+    };
+
+    command_buffer = clCreateCommandBufferKHR(1, &queue, props, nullptr);
+
+    cl_program program =
+        clCreateProgramWithSource(context, 1, &kernelString, nullptr, &error);
+    test_error(error, "Unable to create program");
+
+    error = clBuildProgram(program, 1, &device, nullptr, nullptr, nullptr);
+    test_error(error, "Unable to build program");
+
+    kernel = clCreateKernel(program, "CopyBuffer", &error);
+    test_error(error, "Unable to create kernel");
+
+    return error;
+}
+
+int test_mutable_dispatch_device_query(cl_device_id device, cl_context context,
+                                       cl_command_queue queue, int num_elements)
+{
+    GET_PFN(device, clUpdateMutableCommandsKHR);
+    GET_PFN(device, clCreateCommandBufferKHR);
+    GET_PFN(device, clFinalizeCommandBufferKHR);
+    GET_PFN(device, clCommandNDRangeKernelKHR);
+
+    cl_mutable_dispatch_fields_khr mutable_capabilities;
+    CL_CHECK(clGetDeviceInfo(
+        device, CL_DEVICE_MUTABLE_DISPATCH_CAPABILITIES_KHR,
+        sizeof(mutable_capabilities), &mutable_capabilities, nullptr));
+    if (!(mutable_capabilities & CL_MUTABLE_DISPATCH_ARGUMENTS_KHR))
+    {
+        std::cerr << "Device does not support update arguments to a "
+                     "mutable-dispatch, "
+                     "skipping example.\n";
+        return 0;
+    }
+
+    return TEST_PASS;
+}
+
+int test_mutable_dispatch_command_buffer(cl_device_id device,
+                                         cl_context context,
+                                         cl_command_queue queue,
+                                         int num_elements)
+{
+    GET_PFN(device, clFinalizeCommandBufferKHR);
+    GET_PFN(device, clGetMutableCommandInfoKHR);
+    GET_PFN(device, clCommandNDRangeKernelKHR);
+
+    cl_int error = check_capabilities(device, context);
+
+    cl_command_buffer_khr command_buffer = nullptr;
```

Comment:
Can we use `clCommandBufferWrapper` here, we're not releasing any objects

Reply:
Do we want to move it to some common include file? Right now it is defined in `test_conformance/extensions/cl_khr_command_buffer/command_buffer_test_base.h`  and I don't know if is ok to include it from there.

Reply:
`clCommandBufferWrapper` relies on `CommandBufferTestBase` so I am not sure if we can use it in this context. Alternatively, all of the muable dispatch tests could inherit from `CommandBufferTestBase`.

Reply:
I think mutable-dispatch tests inheriting from `CommandBufferTestBase` is a good design, the mutable-dispatch extension requires many of the functions defined in `CommandBufferTestBase::init_extension_functions()` in order to be useable